### PR TITLE
Calculate metrics of Hierarchical model

### PR DESCRIPTION
**Pull request by sipian**

### CODE REVIEW

File: train_hierarchialXC.py [hunk 1]
```diff
@@ -41,6 +46,29 @@
...
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Calculate the metrics ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+pred_y = []
+actual_y = []
+for x, y in iter(loader):
+    pred_y.append(predictions.toarray())
+    actual_y.append(y.numpy().reshape(-1))
+
+print("Predicted Y values :: Shape : {} , Values: {}", np.array(pred_y).shape, pred_y)
```
Comment (sipian) [inline train_hierarchialXC.py]: for 50 data points
pred_y has shape (50, 50, 159)
actual_y has shape (50, 1, 159)
This is giving erros in calculating the metrics.

Comment (GoodDeeds) [reply train_hierarchialXC.py]: Why are you appending `predictions` `len(loader)` number of times?

Comment (sipian) [reply train_hierarchialXC.py]: Aaah thanks for pointing that out
I'll fix it

Comment (vishwakftw) [reply train_hierarchialXC.py]: Lol

**Comment by sipian** [comment]
@GoodDeeds @vishwakftw 
I have fixed the issue. The metrics are working now

**Comment by vishwakftw** [comment]
You force pushed. ;-)

File: .gitignore [hunk 2]
```diff
@@ -111,3 +111,7 @@ data/
 
 # backup files
 *~
+
+# saved trained model
+*.pt
+*.sav
```
Comment (vishwakftw) [inline .gitignore]: No newline. Add it.

File: train_hierarchialXC.py [hunk 3]
```diff
@@ -1,11 +1,14 @@
 from datetime import datetime
 from argparse import ArgumentParser
 from sklearn import multiclass, svm
+from sklearn.externals import joblib
 
 from extreme_classification.hierarchicalXC import HierarchicalXC
 from extreme_classification.loaders import LibSVMLoader
+from extreme_classification.metrics import precision_at_k, ndcg_score_at_k
```
Comment (vishwakftw) [inline train_hierarchialXC.py]: Can you fix the length ordering?

File: train_hierarchialXC.py [hunk 4]
```diff
@@ -24,6 +27,8 @@
                     help='Option to plot the loss variation over iterations')
 
 # post training arguments
+parser.add_argument('--save_model', action='store_true', default=False,
```
Comment (vishwakftw) [inline train_hierarchialXC.py]: default not required.

File: train_hierarchialXC.py [hunk 5]
```diff
@@ -24,6 +27,8 @@
...
+                    help='Option to save the model completely')
```
Comment (vishwakftw) [inline train_hierarchialXC.py]: "Toggle to save model completely"

File: train_hierarchialXC.py [hunk 6]
```diff
@@ -41,6 +46,25 @@
 
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Train your model ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-my_hierarchical_XC.train(loader, multiclass.OneVsRestClassifier, estimator=svm.SVC())
+my_hierarchical_XC.train(loader, multiclass.OneVsRestClassifier, estimator=svm.SVC(gamma="scale"))
```
Comment (vishwakftw) [inline train_hierarchialXC.py]: Maybe we should add an option for this, but we can leave it to be default for now.

Make it: `estimator=svm.SVC()`.

**Comment by GoodDeeds** [comment]
What is the performance?

File: train_hierarchialXC.py [hunk 7]
```diff
@@ -41,6 +46,25 @@
...
 predictions = my_hierarchical_XC.predict(loader.get_data()[0])
-print(predictions.toarray())
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Calculate the metrics ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+pred_y = predictions.toarray()
+actual_y = []
+for x, y in iter(loader):
+    actual_y.append(y.numpy().reshape(-1))
```
Comment (vishwakftw) [inline train_hierarchialXC.py]: loop is redundant. Make this: `actual_y = loader.get_classes().toarray()` or sth like this.

File: .gitignore (hunk 2 above)
Comment (GoodDeeds) [reply .gitignore]: Out of curiosity, why does it matter?

Comment (vishwakftw) [reply .gitignore]: Actually it doesn't. Scratch this.

**Pull request merged**